### ISSUE TITLE
Weird libglib-2.0 error in dmesg on every boot

### ISSUE THREAD

**Issue by pitsi** [dunst-project/dunst#1134]
### Issue description
I have this error coming up in my dmesg on every single boot. It is not new, it must be there since v1.5 (if not earlier) and the only way to make it go away is to completely uninstall dunst.
```
$ dmesg | grep dunst
[    7.821420] traps: dunst[517] trap int3 ip:7f5e6b10de92 sp:7fff4fc8ea70 error:0 in libglib-2.0.so.0.7400.2[7f5e6b0cf000+8c000]
```

### Installation info

- Version: 1.8.1 
- Install type: package (apt-get install dunst)
- Window manager / Desktop environment: Openbox

### Dunstrc here
I can not create a minimal dunstrc for the issue, so here is my entire dunstrc file. It is basically the default one, with some changes in the font and colors and I created it fresh right after the upgrade to 1.8.1 (straight from 1.5.0). The old file was created around 2016 and I have kept it as a backup.
https://paste.debian.net/hidden/edfe0eeb/

**Comment by pitsi** [comment]
I got the upgrade to v1.9.0 today but the error is still there.

**Comment by pitsi** [comment]
I have never doen a backtrace before, but I will try it. Please be patient :)

**Comment by fwsmit** [comment]
If it always crashes you can do the following:
```
pkill dunst
gdb dunst -ex r -ex bt
```

Also, what distro are you using?

**Comment by fwsmit** [comment]
But I'm not even sure if a trap is a crash. If not, it's not as easy to generate a backtrace

**Comment by pitsi** [comment]
It does not crash at all. It is just that weird message it pops on every in dmesg.
I am on debian testing/unstable x64.

**Comment by pitsi** [comment]
Ok here goes my first try with gdb.
```
Reading symbols from dunst...
(No debugging symbols found in dunst)
Starting program: /usr/bin/dunst 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff6d786c0 (LWP 4003)]
[New Thread 0x7ffff65776c0 (LWP 4004)]
[New Thread 0x7ffff57ff6c0 (LWP 4005)]
[Thread 0x7ffff65776c0 (LWP 4004) exited]
[New Thread 0x7ffff65776c0 (LWP 4016)]
[New Thread 0x7ffff4ffe6c0 (LWP 4017)]
[New Thread 0x7fffe7fff6c0 (LWP 4021)]
[Thread 0x7fffe7fff6c0 (LWP 4021) exited]
[New Thread 0x7fffe7fff6c0 (LWP 4022)]
[New Thread 0x7fffe77fe6c0 (LWP 4023)]
[New Thread 0x7fffe7fff6c0 (LWP 4024)]
[Thread 0x7fffe7fff6c0 (LWP 4022) exited]
[New Thread 0x7fffe6ffd6c0 (LWP 4025)]
[Thread 0x7fffe7fff6c0 (LWP 4024) exited]
[Thread 0x7fffe6ffd6c0 (LWP 4025) exited]
[New Thread 0x7fffe6ffd6c0 (LWP 4026)]
[New Thread 0x7fffe7fff6c0 (LWP 4027)]
[Thread 0x7fffe77fe6c0 (LWP 4023) exited]
[New Thread 0x7fffe6ffd6c0 (LWP 4028)]
[Thread 0x7fffe6ffd6c0 (LWP 4026) exited]
[New Thread 0x7fffe77fe6c0 (LWP 4029)]
[Thread 0x7fffe6ffd6c0 (LWP 4028) exited]
[Thread 0x7fffe7fff6c0 (LWP 4027) exited]
[Thread 0x7fffe77fe6c0 (LWP 4029) exited]
[Thread 0x7ffff65776c0 (LWP 4016) exited]
```
I sent a "notify-send something" to see if it works and it was working. After all that I pressed ctrl+c and then q to exit gdb.

**Comment by fwsmit** [comment]
There is nothing useful in this output. Does the issue even happen when running dunst from the command line? If not, you'll have to look at how exactly dunst is started and what is different about starting that way.

**Comment by pitsi** [comment]
I thought so :(
The issue seems to happen only on boot and not when running dunst again (after killing it). Sadly, I can not find how it is started on boot or login. It has a systemd service, but it is not active, it is not started from /etc/xdg/autostart/ (like wbar) and I do not start it manually via ~/config/openbox/autostart (like conky).

**Comment by bynect** [comment]
maybe dunst is started too soon? can you change the order of the processes?

**Comment by pitsi** [comment]
How can I know when it is started? There is a dunst.service file, but it is not even enabled as it seems.
```
$ systemctl status dunst.service
Unit dunst.service could not be found.
```

**Comment by pitsi** [comment]
It may be dbus, because it has some dbus related file as you can see here
https://packages.debian.org/sid/amd64/dunst/filelist

As for dunst being the only notification daemon, it probably is because I have none of the remaining packages listed here that can provide A notification daemon
https://packages.debian.org/sid/notification-daemon

**Comment by bynect** [comment]
It seems like some other user in the internet incurred in a sigtrap sent by debian libglib.
https://stackoverflow.com/questions/61816297/what-is-int-3-really-supposed-to-do
https://stackoverflow.com/questions/63340398/trap-int3-error-in-libglib-debian-eclipse
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=935889

Now, I suspect you are using an x86 system, because it seems for those systems libglib uses traps instead of abort.

My hypothesis: during the boot process some program (maybe even dbus itself) is starting dunst too early, so dunst is erroring out in some unknown way that is causing an error in libglib. Or maybe libglib is corrupted/unilitialized.

I would say you can do a diskcheck, reinstall libglib competely and also dunst to the latest version. Then if the error is not solved, figure out where dunst is being started

**Comment by pitsi** [comment]
Libglib has taken a couple of updates since then, like dunst which is now on 1.9.2. In fact, it is now on 2.78.**4** and the only thing different in dmesg (excluding the memory addresses) is the name of the lib
```
$ dmesg | grep dunst
[   13.613972] traps: dunst[631] trap int3 ip:7f559fcb1c79 sp:7ffe2a297d20 error:0 in libglib-2.0.so.0.7800.4[7f559fc6d000+99000]
```
And it seems that libglib is no ordinary library, because dozens of packages depend on it on my (simplistic) setup. All 3 of my browsers, gtk2/3 apps and libraries, qt5/6 apps and libraries and even apps with no ui, like mpv or mplayer, depend on it. And yet, no other app pops anything similar in dmesg so as to indicate it is corrupted.

As for quering dbus for dunst, it gives a similar error to systemd about dunst's status 
```
$ busctl status dunst.service
Failed to get credentials: No such device or address
$ sudo busctl status dunst.service
Failed to get credentials: No such device or address
```

**Comment by bynect** [comment]
> Libglib has taken a couple of updates since then, like dunst which is now on 1.9.2. In fact, it is now on 2.78.**4** and the only thing different in dmesg (excluding the memory addresses) is the name of the lib
> ```
> $ dmesg | grep dunst
> [   13.613972] traps: dunst[631] trap int3 ip:7f559fcb1c79 sp:7ffe2a297d20 error:0 in libglib-2.0.so.0.7800.4[7f559fc6d000+99000]
> ```
> And it seems that libglib is no ordinary library, because dozens of packages depend on it on my (simplistic) setup. All 3 of my browsers, gtk2/3 apps and libraries, qt5/6 apps and libraries and even apps with no ui, like mpv or mplayer, depend on it. And yet, no other app pops anything similar in dmesg so as to indicate it is corrupted.
> 
> As for quering dbus for dunst, it gives a similar error to systemd about dunst's status 
> ```
> $ busctl status dunst.service
> Failed to get credentials: No such device or address
> $ sudo busctl status dunst.service
> Failed to get credentials: No such device or address
> ```
> 

This is really weird. Is my assumption of 32 bit libglib right?

I can think of two thing at the moment:
Either the version of dunst is installed incorrectly (which is weird considering debian) or again dunst is started in an improper way during boot.

I would say, try to remove the debian version and compile and install from git and see if still persist with manual installation. If that's the case we have to find out who and when starts dunst

Also, could you send the surrounding context of the dmesg error?

**Comment by bynect** [comment]
Also libglib is like an extra standard library for c made by gtk, so it's normal that many program depend on it. Verifying the installation wouldn't hurt though

**Comment by pitsi** [comment]
It's 64 bit. Call me cocky, but there is no 32bit lib on my system.
```
$ file /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7800.4
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7800.4: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=494f1601eab85c409c692018c500f17028ec31c2, stripped
```
And here is a full dmesg, although there is only one line for dunst 
https://paste.debian.net/hidden/cb01eaf9/

As for compiling it on my own... yea this is not a task for me, sorry :(

**Comment by bynect** [comment]
> It's 64 bit. Call me cocky, but there is no 32bit lib on my system.
> ```
> $ file /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7800.4
> /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7800.4: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=494f1601eab85c409c692018c500f17028ec31c2, stripped
> ```
> And here is a full dmesg, although there is only one line for dunst 
> https://paste.debian.net/hidden/cb01eaf9/
> 
> As for compiling it on my own... yea this is not a task for me, sorry :(

I see. Online I found that the trap signal is used instead of abort on 32 bit, but on x64 am not too sure.

To compile yourself is fairly simple, you have just to download the repository (either git clone or the zip) and run `make install`

**Comment by pitsi** [comment]
I know that fairly simple way. Before that "make install" part, I must also have all build dependencies available or it won't build, and that is annoying for me. Then I have to uninstall 
Anyway, I will look for a way to check how dbus or anything loads it at the start. If I don't find anything helpful, I will close the issue and let dunst be, as long as it still works..

**Comment by bynect** [comment]
Looking back to it, maybe dunst was being started before the  wm/de was properly initialized. #1174 presents a segfault when dunst is started before the wm. Maybe it is related?

In general dunst should be launched only after dbus and a xorg session. So maybe you have to change the order of process or disable dunst being started during boot?

**Comment by pitsi** [comment]
How can I do that? Because 
a) we are in the systemd era and such "this starts after that" incidents are now taken care by systemd with the "After=" parameter, e.g. transmission-daemon starts after networking because of this
`After=network.target`
b) dunst does not start as a service here, as I mentioned above

**Comment by bynect** [comment]
> How can I do that? Because 
> a) we are in the systemd era and such "this starts after that" incidents are now taken care by systemd with the "After=" parameter, e.g. transmission-daemon starts after networking ONLY because of this bit
> `After=network.target`
> b) dunst does not start as a service here, as I mentioned above
> 
> Dbus is started as a service and xorg is started by lightdm, which in turn is started as a service too.

Make a dunst service and make it depend on dbus/xorg. We have a systemd package already written in the repo. You can try using it. 

However if you it persist you have to find who starts dunst

**Comment by pitsi** [comment]
I think I will do the second option first. So, assuming dbus starts it, how can I tell when it starts or stop it temporarily? Stopping dbus completely is out of the question, as it is a core component of the entire os and many other services and apps will fail.

**Comment by bynect** [comment]
> I think I will do the second option first. So, assuming dbus starts it, how can I tell when it starts or stop it temporarily? Stopping dbus completely is out of the question, as it is a core component of the entire os and many other services and apps will fail.

I am not sure that dbus is the one starting dunst in the first place. To help you more you should get a core dump from dunst (for example with systemd coredump)

I think trying the systemd service file is fairly easy and non breaking, could you try it just in case?

**Comment by pitsi** [comment]
I made a major discovery! Dunst IS started as a systemd service, but as a USER service, so I assume this is what you want.
```
$ systemctl status dunst --user
● dunst.service - Dunst notification daemon
     Loaded: loaded (/usr/lib/systemd/user/dunst.service; enabled; preset: enabled)
     Active: active (running) since Sun 2024-03-03 17:11:58 EET; 1h 44min ago
       Docs: man:dunst(1)
   Main PID: 1590 (dunst)
      Tasks: 4 (limit: 4665)
     Memory: 3.4M (peak: 3.9M)
        CPU: 1.126s
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/dunst.service
             └─1590 /usr/bin/dunst

Mar 03 17:11:58 mitsos systemd[612]: Starting dunst.service - Dunst notification daemon...
Mar 03 17:11:58 mitsos systemd[612]: Started dunst.service - Dunst notification daemon.
```

**Comment by bynect** [comment]
good to know! Does systemd allow you to see a stack trace / core dump when the crash occurs? Also is the service properly configured to start after the wm/de?

**Comment by pitsi** [comment]
Do these help?
```
$ cat /usr/lib/systemd/user/dunst.service
[Unit]
Description=Dunst notification daemon
Documentation=man:dunst(1)
PartOf=graphical-session.target

[Service]
Type=dbus
BusName=org.freedesktop.Notifications
ExecStart=/usr/bin/dunst
```
```
$ cat /usr/share/dbus-1/services/org.knopwob.dunst.service
[D-BUS Service]
Name=org.freedesktop.Notifications
Exec=/usr/bin/dunst
SystemdService=dunst.service
```
I am still trying to figure out how to ask systemd for a coredump.

**Comment by bynect** [comment]
The systemd service file is the same in the repo so it should work.

> am still trying to figure out how to ask systemd for a coredump.

https://www.freedesktop.org/software/systemd/man/latest/systemd-coredump.html

If we can pinpoint the crash to a precise stacktrace it would be ideal for debugging 👍🏻

**Comment by bynect** [comment]
Another user resolved a similar issue here:
https://github.com/dunst-project/dunst/issues/1340#issuecomment-2066733836

**Comment by pitsi** [comment]
Please excuse my ignorance, but how to I disable a systemd service which runs for the user? At first I thought it of
`systemd disable dunst --user`
but then I saw this thread on reddit where the ts talks about deleting files and I really don't want to do it!
https://old.reddit.com/r/linuxquestions/comments/mzgpiq/how_do_i_disable_a_systemd_user_service/

**Comment by bynect** [comment]
I think disabling the systemd service and starting dunst manually in your xinitrc (or equivalent if you are using a DE) would be fine.
I am not a systemd expert but I think that either `systemd disable` or `systemd mask` should work. Anyway they only modify symlinks and the unit files are untouched so don't worry.

**Comment by pitsi** [comment]
I disabled the systemd service the way I said above, I ran `notify-send asdf` and a notification appeared on my screen. As it seems dunst did start on boot, and even showed the same error in dmesg.
`[   13.776925] traps: dunst[636] trap int3 ip:7f583e350c79 sp:7fffbb6f6da0 error:0 in libglib-2.0.so.0.7800.4[7f583e30c000+99000]`

Also, this last line showed up on systemctl status when I sent the notification
```
$ systemctl status dunst --user
● dunst.service - Dunst notification daemon
     Loaded: loaded (/usr/lib/systemd/user/dunst.service; enabled; preset: enabled)
     Active: active (running) since Sat 2024-04-20 17:46:14 EEST; 21min ago
       Docs: man:dunst(1)
   Main PID: 882 (dunst)
      Tasks: 5 (limit: 4665)
     Memory: 6.7M (peak: 7.3M)
        CPU: 184ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/dunst.service
             └─882 /usr/bin/dunst

Apr 20 17:46:14 mitsos systemd[617]: Starting dunst.service - Dunst notification daemon...
Apr 20 17:46:14 mitsos systemd[617]: Started dunst.service - Dunst notification daemon.
Apr 20 18:07:28 mitsos dunst[882]: CRITICAL: Source ID 20 was not found when attempting to remove it
```

**Comment by bynect** [comment]
> I disabled the systemd service the way I said above, I ran `notify-send asdf` and a notification appeared on my screen. As it seems dunst did start on boot, and even showed the same error in dmesg.
> `[   13.776925] traps: dunst[636] trap int3 ip:7f583e350c79 sp:7fffbb6f6da0 error:0 in libglib-2.0.so.0.7800.4[7f583e30c000+99000]`
> 
> Also, this last line showed up on systemctl status when I sent the notification
> ```
> $ systemctl status dunst --user
> ● dunst.service - Dunst notification daemon
>      Loaded: loaded (/usr/lib/systemd/user/dunst.service; enabled; preset: enabled)
>      Active: active (running) since Sat 2024-04-20 17:46:14 EEST; 21min ago
>        Docs: man:dunst(1)
>    Main PID: 882 (dunst)
>       Tasks: 5 (limit: 4665)
>      Memory: 6.7M (peak: 7.3M)
>         CPU: 184ms
>      CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/dunst.service
>              └─882 /usr/bin/dunst
> 
> Apr 20 17:46:14 machine systemd[617]: Starting dunst.service - Dunst notification daemon...
> Apr 20 17:46:14 machine systemd[617]: Started dunst.service - Dunst notification daemon.
> Apr 20 18:07:28 machine dunst[882]: CRITICAL: Source ID 20 was not found when attempting to remove it
> ```
> 

From systemd it seems like the service is still enabled. Maybe you can try masking it?

About that message, it was fixed in the latest release. However it should not cause problem either way

**Comment by fwsmit** [comment]
If dunst is not running, it will be started by DBus. So even if you disable it, it can start. I don't see what the problem is though

**Comment by pitsi** [comment]
The problem is that I want to disable its systemd start and start it via autostart, for troubleshooting. Obviously disabling dbus is a big no, because a ton of other stuff depends on it.

**Comment by pitsi** [comment]
I finally managed to disable it by masking it like so, because no other masking way worked as you can see.
```
$ sudo systemctl mask dunst
Unit dunst.service does not exist, proceeding anyway.
Created symlink /etc/systemd/system/dunst.service → /dev/null.

$ sudo systemctl unmask dunst
Removed "/etc/systemd/system/dunst.service".

$ sudo systemctl mask dunst --user
Failed to connect to bus: No medium found

$ systemctl --user mask dunst
Created symlink /home/user/.config/systemd/user/dunst.service → /dev/null.
```
And then I added it to my ~/.config/openbox/autostart like so
```
dunst &
sleep 3 &
```
but still nothing.
```
$ dmesg | grep dunst
[  124.267485] traps: dunst[1129] trap int3 ip:7f386b3e4c79 sp:7fff3dadf1e0 error:0 in libglib-2.0.so.0.7800.4[7f386b3a0000+99000]
```

**Comment by fwsmit** [comment]
For troubleshooting you can just do pkill dunst && dunst &

**Comment by bynect** [comment]
> I just tried rebooting and the error did not show up on dmesg. Now what?

I guess it is resolved now. Does dunst actually work if you send a notification?

**Comment by pitsi** [comment]
Yes it does. We can conclude that the error shows up when dunst is started via its systemd service file because the lib is not loaded yet.I think I will leave it that way, provided that debian's dunst update to a newer version won't break it. 
For now, debian is still at 1.9.2 :(

**Comment by bynect** [comment]
> Yes it does. We can conclude that the error shows up when dunst is started via its systemd service file because the lib is not loaded yet.I think I will leave it that way, provided that debian's dunst update to a newer version won't break it. 
> For now, debian is still at 1.9.2 :(

Glad it worked. Maybe in the future we can tweak the systemd file to prevent this problem at the root

**Comment by dlundqvist** [comment]
Info and backtrace from my system during boot:
```
Sep 11 07:51:06 neptune systemd[1541]: Starting dunst.service - Dunst notification daemon...
Sep 11 07:51:06 neptune dunst[1561]: WARNING: Cannot open X11 display.
Sep 11 07:51:06 neptune dunst[1561]: ERROR: [  get_x11_output:0070] Couldn't initialize X11 output. Aborting...
Sep 11 07:51:06 neptune systemd-coredump[1583]: [🡕] Process 1561 (dunst) of user NNNN dumped core.
                                                
                                                Module libblkid.so.1 from deb util-linux-2.40.2-7.amd64
                                                Module libatomic.so.1 from deb gcc-14-14.2.0-3.amd64
                                                Module libmount.so.1 from deb util-linux-2.40.2-7.amd64
                                                Stack trace of thread 1561:
                                                #0  0x00007f98da36769a g_logv (libglib-2.0.so.0 + 0x6569a)
                                                #1  0x00007f98da367903 g_log (libglib-2.0.so.0 + 0x65903)
                                                #2  0x000055a3c26af618 n/a (dunst + 0x1d618)
                                                #3  0x000055a3c26a5098 n/a (dunst + 0x13098)
                                                #4  0x000055a3c26a7ae1 n/a (dunst + 0x15ae1)
                                                #5  0x00007f98d9fb1dba __libc_start_call_main (libc.so.6 + 0x29dba)
                                                #6  0x00007f98d9fb1e75 __libc_start_main_impl (libc.so.6 + 0x29e75)
                                                #7  0x000055a3c26a26a1 n/a (dunst + 0x106a1)
                                                
                                                Stack trace of thread 1578:
                                                #0  0x00007f98da081d2f __GI___poll (libc.so.6 + 0xf9d2f)
                                                #1  0x00007f98da55be4d g_socket_condition_timed_wait (libgio-2.0.so.0 + 0xa8e4d)
                                                #2  0x00007f98da55cf7a n/a (libgio-2.0.so.0 + 0xa9f7a)
                                                #3  0x00007f98da5376e5 g_input_stream_read (libgio-2.0.so.0 + 0x846e5)
                                                #4  0x00007f98da506bc0 n/a (libgio-2.0.so.0 + 0x53bc0)
                                                #5  0x00007f98da507224 g_buffered_input_stream_fill (libgio-2.0.so.0 + 0x54224)
                                                #6  0x00007f98da50f0e6 g_data_input_stream_read_line (libgio-2.0.so.0 + 0x5c0e6)
                                                #7  0x00007f98da5baee9 n/a (libgio-2.0.so.0 + 0x107ee9)
                                                #8  0x00007f98da5bbc22 n/a (libgio-2.0.so.0 + 0x108c22)
                                                #9  0x00007f98da5cda79 n/a (libgio-2.0.so.0 + 0x11aa79)
                                                #10 0x00007f98da505690 n/a (libgio-2.0.so.0 + 0x52690)
                                                #11 0x00007f98da56d171 n/a (libgio-2.0.so.0 + 0xba171)
                                                #12 0x00007f98da38f732 n/a (libglib-2.0.so.0 + 0x8d732)
                                                #13 0x00007f98da38f161 n/a (libglib-2.0.so.0 + 0x8d161)
                                                #14 0x00007f98da014732 start_thread (libc.so.6 + 0x8c732)
                                                #15 0x00007f98da08f2b8 __clone3 (libc.so.6 + 0x1072b8)
                                                
                                                Stack trace of thread 1576:
                                                #0  0x00007f98da08d279 syscall (libc.so.6 + 0x105279)
                                                #1  0x00007f98da38ea84 g_cond_wait (libglib-2.0.so.0 + 0x8ca84)
                                                #2  0x00007f98da3263fb n/a (libglib-2.0.so.0 + 0x243fb)
                                                #3  0x00007f98da38f402 n/a (libglib-2.0.so.0 + 0x8d402)
                                                #4  0x00007f98da38f161 n/a (libglib-2.0.so.0 + 0x8d161)
                                                #5  0x00007f98da014732 start_thread (libc.so.6 + 0x8c732)
                                                #6  0x00007f98da08f2b8 __clone3 (libc.so.6 + 0x1072b8)
                                                
                                                Stack trace of thread 1577:
                                                #0  0x00007f98da082266 __GI_ppoll (libc.so.6 + 0xfa266)
                                                #1  0x00007f98da35eaec n/a (libglib-2.0.so.0 + 0x5caec)
                                                #2  0x00007f98da35f180 g_main_context_iteration (libglib-2.0.so.0 + 0x5d180)
                                                #3  0x00007f98da35f1d1 n/a (libglib-2.0.so.0 + 0x5d1d1)
                                                #4  0x00007f98da38f161 n/a (libglib-2.0.so.0 + 0x8d161)
                                                #5  0x00007f98da014732 start_thread (libc.so.6 + 0x8c732)
                                                #6  0x00007f98da08f2b8 __clone3 (libc.so.6 + 0x1072b8)
                                                ELF object binary architecture: AMD x86-64
```

Another when system is shutting down:
```
Sep 10 18:54:22 neptune dunst[1720]: X connection to :0 broken (explicit kill or server shutdown).
Sep 10 18:54:22 neptune systemd[1542]: dunst.service: Main process exited, code=exited, status=1/FAILURE
Sep 10 18:54:22 neptune systemd[1542]: dunst.service: Failed with result 'exit-code'.```

It seems `dunst` is not very tolerant about not being able to connect to X server, nor losing the connection. Could this be made less severe?

**Comment by pitsi** [comment]
I checked my dmesg again, now that github's notification brought it to my attention, and the issue is still there on 1.11!
How can I check if it runs as a systemd service or something else? It is not on my openbox's autostart file though
```
$ dunst --version
Dunst - A customizable and lightweight notification-daemon 1.11.0 (2024-04-15)
$ dmesg | grep dunst
[   10.687199] traps: dunst[651] trap int3 ip:7fcf32b6b69a sp:7ffe331ac360 error:0 in libglib-2.0.so.0.8200.0[6569a,7fcf32b24000+9f000]
```

**Comment by bynect** [comment]
> It seems `dunst` is not very tolerant about not being able to connect to X server, nor losing the connection. Could this be made less severe?

I guess we  could change the abort to a exit(1) or something. But the best practice would be to start/kill dunst when the X env is up

**Comment by dlundqvist** [comment]
> > It seems `dunst` is not very tolerant about not being able to connect to X server, nor losing the connection. Could this be made less severe?
> 
> I guess we could change the abort to a exit(1) or something. But the best practice would be to start/kill dunst when the X env is up

Thank you for considering this. I don't explicitly start it so my guess is that something early in the login process wants to send a notification but X is not yet. Do you want me to investigate this further?

**Comment by bynect** [comment]
> Thank you for considering this. I don't explicitly start it so my guess is that something early in the login process wants to send a notification but X is not ready yet. Do you want me to investigate this further?

if you can I would be grateful.

a couple thing ideas to test:
* check all initialization programs: systemd (or other init), login managers, startx, etc
* if you have the possibility start your system without graphical interface and check if the error happens even in a purely console setting

**Comment by tripplehelix** [comment]
I'm having the same issue and would like to answer your questions;

The errors:

```
[   10.436682] traps: dunst[1755] trap int3 ip:7f4188c5fdda sp:7ffc55cdfe20 error:0 in libglib-2.0.so.0.8400.0[65dda,7f4188c18000+a1000]
[   10.456060] traps: dunst[1802] trap int3 ip:7fa908a00dda sp:7ffc8a248e80 error:0 in libglib-2.0.so.0.8400.0[65dda,7fa9089b9000+a1000]
[   15.661102] usb 1-4: reset full-speed USB device number 3 using xhci_hcd
[   15.751737] traps: dunst[1967] trap int3 ip:7f50ba43cdda sp:7fff8ad19530 error:0 in libglib-2.0.so.0.8400.0[65dda,7f50ba3f5000+a1000]
[   15.944996] usb 1-4: reset full-speed USB device number 3 using xhci_hcd
[   21.051446] traps: dunst[1987] trap int3 ip:7f75525f6dda sp:7ffc4ade6d40 error:0 in libglib-2.0.so.0.8400.0[65dda,7f75525af000+a1000]
[   26.212207] traps: dunst[2014] trap int3 ip:7fdee3461dda sp:7fffea31d5a0 error:0 in libglib-2.0.so.0.8400.0[65dda,7fdee341a000+a1000]
```

Have no issues with other Systemd services. However, these are a small selection of dunst messages from journalctl:

```
Mar 28 15:11:46 laptop kernel: traps: dunst[1802] trap int3 ip:7fa908a00dda sp:7ffc8a248e80 error:0 in libglib-2.0.so.0.8400.0[65dda,7fa9089b9000+a1000]
Mar 28 15:11:46 laptop systemd[1576]: dunst.service: Main process exited, code=killed, status=5/TRAP
Mar 28 15:11:46 laptop systemd[1576]: dunst.service: Failed with result 'signal'.
Mar 28 15:11:51 laptop kernel: traps: dunst[1967] trap int3 ip:7f50ba43cdda sp:7fff8ad19530 error:0 in libglib-2.0.so.0.8400.0[65dda,7f50ba3f5000+a1000]
Mar 28 15:11:51 laptop systemd[1576]: dunst.service: Scheduled restart job, restart counter is at 2.
Mar 28 15:11:51 laptop dunst[1967]: WARNING: Cannot open X11 display.
Mar 28 15:11:51 laptop dunst[1967]: ERROR: [  get_x11_output:0070] Couldn't initialize X11 output. Aborting...
Mar 28 15:11:51 laptop systemd[1576]: Started dunst.service - System notifications.
Mar 28 15:11:51 laptop systemd[1576]: dunst.service: Main process exited, code=killed, status=5/TRAP
Mar 28 15:11:51 laptop systemd[1576]: dunst.service: Failed with result 'signal'.
Mar 28 15:11:56 laptop systemd[1576]: dunst.service: Scheduled restart job, restart counter is at 3.
Mar 28 15:11:56 laptop systemd[1576]: Started dunst.service - System notifications.
Mar 28 15:11:56 laptop dunst[1987]: WARNING: Cannot open X11 display.
Mar 28 15:11:56 laptop dunst[1987]: ERROR: [  get_x11_output:0070] Couldn't initialize X11 output. Aborting...
Mar 28 15:11:56 laptop kernel: traps: dunst[1987] trap int3 ip:7f75525f6dda sp:7ffc4ade6d40 error:0 in libglib-2.0.so.0.8400.0[65dda,7f75525af000+a1000]
Mar 28 15:11:56 laptop systemd[1576]: dunst.service: Main process exited, code=killed, status=5/TRAP
Mar 28 15:11:56 laptop systemd[1576]: dunst.service: Failed with result 'signal'.
Mar 28 15:12:02 laptop systemd[1576]: dunst.service: Scheduled restart job, restart counter is at 4.
Mar 28 15:12:02 laptop systemd[1576]: Started dunst.service - System notifications.
Mar 28 15:12:02 laptop dunst[2014]: WARNING: Cannot open X11 display.
Mar 28 15:12:02 laptop dunst[2014]: ERROR: [  get_x11_output:0070] Couldn't initialize X11 output. Aborting...
Mar 28 15:12:02 laptop systemd[1576]: dunst.service: Main process exited, code=killed, status=5/TRAP
Mar 28 15:12:02 laptop systemd[1576]: dunst.service: Failed with result 'signal'.
Mar 28 15:12:02 laptop kernel: traps: dunst[2014] trap int3 ip:7fdee3461dda sp:7fffea31d5a0 error:0 in libglib-2.0.so.0.8400.0[65dda,7fdee341a000+a1000]
Mar 28 15:12:07 laptop systemd[1576]: dunst.service: Scheduled restart job, restart counter is at 5.
Mar 28 15:12:07 laptop systemd[1576]: Started dunst.service - System notifications.
Mar 28 15:12:07 laptop dunst[2545]: WARNING: Could not find theme breeze
Mar 28 15:12:07 laptop dunst[2545]: WARNING: Icon 'nm-signal-75' not found in themes
```

ADDING ANSWER TO QUESTION TWO ONCE I RESTART

**Comment by pitsi** [comment]
Can you try disabling dunst as a systemd service and starting it on login (as user), like I did above?
Truth be told, dust's upgrades here reenabled the systemd service, so I have the error again, but I no longer bother.

**Comment by tripplehelix** [comment]
Could I ask the reasoning for disabling the systemd service?

**Comment by pitsi** [comment]
As mentioned above, you disable the systemd service and dunst loads later so that the library in question is loaded before dunst and it does not complain.

**Comment by tripplehelix** [comment]
If that's the case, I'll set a startup delay on the systemd service.
